### PR TITLE
Upgrade trac 1.0.1

### PR DESCRIPTION
**Pull request by dreid**
Attempt to automate the upgrade to to trac 1.0.1

This goes along with a t-web pull request.

### CODE REVIEW

File: fabfile.py [hunk 1]
```diff
@@ -119,5 +119,24 @@ def task_restore(self, localfile, restoreDb=True):
                         postgres.restoreFromPath('trac', temp)
 
 
+    def task_upgrade10(self):
+        """
+        potato
```
Comment (alex) [inline fabfile.py]: indeed

File: fabfile.py [hunk 2]
```diff
@@ -119,5 +119,24 @@ def task_restore(self, localfile, restoreDb=True):
...
+        with settings(user=self.serviceUser):
+            self.update()
+
+            run("/bin/mv ~/attachments ~/attachments.old")
+            run("/bin/ln -nsf ~/attachments.old {}/trac-env/attachments".format(self.configDir))
+
+            run("/bin/mkdir ~/attachments")
+            run("/bin/mkdir {}/trac-env/files".format(self.configDir))
+            run("/bin/ln -nsf ~/attachments {}/trac-env/files/attachments".format(
+                self.configDir))
```
Comment (tomprince) [inline fabfile.py:145]: Huh?

File: trac-env/conf/trac.ini [hunk 3]
```diff
@@ -129,7 +130,7 @@ style = compact
 trunks = trunk
 
 [spam-filter]
```
Comment (tomprince) [inline trac-env/conf/trac.ini:133]: There are some things in this section that should be moved to a separate file, if possible. I think there are also some changes on dornkirk that were made through the admin interface. :/

File: fabfile.py (hunk 2 above)
Comment (tomprince) [inline fabfile.py:145]: Do we need both `attachments` and `attahcments.old`? In any case, the dump/restore will need to be adjusted.

File: fabfile.py (hunk 1 above)
Comment (tomprince) [inline fabfile.py]: Can the need to do this be detected and added to update? Also, `install` will need to be updated too.

File: fabfile.py (hunk 2 above)
Comment (dreid) [inline fabfile.py:145]: I did this little dance so trac could perform it's attachments upgrade, and dump/restore wouldn't need to be changed.

File: trac-env/conf/trac.ini (hunk 3 above)
Comment (dreid) [inline trac-env/conf/trac.ini:133]: I do not think that should be done in this pull request, though a reasonable prerequisite to upgrading would be merging dornkirk's trac.ini with what exists in braid.

**Pull request merged**